server/services/offer.service.js

```diff
@@ -1,15 +1,36 @@
 import mongoose from 'mongoose';
+import format from 'date-fns/format';
 import Offer from '../models/offer.model';
 import { ErrorHandler } from '../helpers/errorHandler';
 import httpStatus from '../helpers/httpStatus';
 import { OFFER_STATUS } from '../helpers/constants';
 import { getUserById } from './user.service';
 import { getEnquiryById, approveEnquiry } from './enquiry.service';
+import { getPropertyById } from './property.service';
 
 const { ObjectId } = mongoose.Types.ObjectId;
 
 export const getOfferById = async (id) => Offer.findById(id).select();
 
+export const generateReferenceCode = async (propertyId) => {
+  const property = await getPropertyById(propertyId);
+  const initials = property.name
```

Comment:
This should be extracted into its own function so that it is easier to know what it does, reuse later if needed and if possible make it easier for us to write test for it.

Reply:
extracting this to its own function would make us query the database twice for the same thing due to line 28

Reply:
I am sorry, I dont understand how extracting this as a function will affect the database. It has nothing to do with the database.

 ```suggestion
  const initials = getPropertyInitial(property.name)
```

```
const getPropertyInitial = (name) => name 
     .match(/\b(\w)/g)
    .join('')
    .toUpperCase();
```

Reply:
ohhh. I was thought process was very different. Would fix it

Reply:
Extracting it in a different function makes the code easier to read for anyone that wants to work on the code later in the future. All of this will not make sense to you now since you are working on the code from scratch, if you are working on legacy codes (i.e. codes that have been written by different people), all of this will make more sense to you.